docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+The format standard of container image encryption is compatible with https://github.com/containers/ocicrypt.
+
+The format standard of container image signing is compatible with https://github.com/containers/image.
+
+### Image encryption
+
+Image encryption is based on layer granularity. Users can selectively encrypt layers containing private data or choose not to encrypt any layers. The high-level process of encrypting an image layer is as follows:
+
+1. Dynamically generate a random symmetric key.
+
+2. Encrypt container image layer with symmetric key.
+
+3. Encrypt the symmetric key with the user's private key.
+
+4. Write the symmetric key encrypted by the user's private key into the container image manifest as the content of the annotation of the container image layer.
+
+The above process needs to be carried out according to the container image encryption format specification defined in [containers/ocicrypt](https://github.com/containers/ocicrypt). In the confidential container V1 architecture, image-rs calls the ocicrypt-rs's APIs for specific implementation.
+
+Details are briefly described below:
+
+After dynamically generating the random symmetric key and encrypting the image layer, two special JSON structures need to be generated, One is used to record public encrypted information (such as encryption algorithm), which is called `PublicLayerBlockCipherOptions`, and the other is used to record secret information (such as symmetric key used for encryption), which is called `PrivateLayerBlockCipherOptions`. The formats are as follows:
+
+- `PublicLayerBlockCipherOptions`:
+
+  ```json
+  {
+      "cipher": "AES_256_CTR_HMAC_SHA256",
+      "hmac": “M0M5OTA5QUZFQzI1MzU0RDU1MURBR…”
+      "cipheroptions": {}
+  }
+  ```
+
+- `PrivateLayerBlockCipherOptions`:
+
+  ```json
+  {
+      "symkey": "54kiln1USEaKnlYhKdz+aA==",
+      "cipheroptions": {
+          "nonce": "AdcRPTAEhXx6uwuYcOquNA=="
+      }
+  }
+  ```
+
+Then, encrypt the `PrivateLayerBlockCipherOptions` with the user's private key. Under the design of confidential containers V1, when decrypting the image, the decryption of `PrivateLayerBlockCipherOptions` needs to be carried out through the key provider protocol with the attestation-agent. Therefore, when encrypting the image, the `PrivateLayerBlockCipherOptions` also needs to be encrypted through the key provider protocol, Moreover, the name of the key provider program used in encryption needs to be consistent with that used in decryption. Under the design of confidential containers V1, this name is uniformly specified as "attestation-agent".
+
+The key provider service program encrypts the `PrivateLayerBlockCipherOptions` with the user's OpenPGP key. After that, it will perform Base64 encoding and then package it into a JSON structure. Under the confidential container system, it is called 'annotation packet' to record some additional information, such as the ID of the user's private key, for example:
```

Comment:
This is the first reference to OpenPGP. Do we want to make any reference to OpenPGP (arguably an implementation detail)?

Is there any necessary relation of that key with the user's private key in step 3 line 21 above? If so, use the same terminology. If not, let's explain cases where the keys could be different and how it would work.

Reply:
I have added the link of OpenPGP RFC. The current signature system only supports OpenPGP, but encryption can allow various standards.

In the future, when we implement this scheme, we may consider unified management of the signature key and encryption key on the owner side, but whether to unify the two keys may not be the focus of this document.